bindings/node/test/autoEncrypter.test.js

```diff
@@ -538,4 +608,52 @@ describe('AutoEncrypter', function () {
       });
     });
   });
+
+  describe('CSFLE shared library', function () {
+    it('should load a shared library by specifying its path', function () {
+      const client = new MockClient();
+      this.mc = new AutoEncrypter(client, {
+        keyVaultNamespace: 'admin.datakeys',
+        logger: () => {},
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        },
+        extraOptions: {
+          csflePath: sharedLibraryStub
+        }
+      });
+
+      expect(this.mc._mongocryptdManager).to.equal(undefined);
+      expect(this.mc._mongocryptdClient).to.equal(undefined);
+      expect(this.mc.csfleVersionInfo).to.deep.equal({
+        // eslint-disable-next-line no-undef
+        version: BigInt(0x000600020001000),
+        versionStr: 'stubbed-mongo_csfle'
+      });
+    });
+
+    it('should load a shared library by specifying a search path', function () {
+      const client = new MockClient();
+      this.mc = new AutoEncrypter(client, {
+        keyVaultNamespace: 'admin.datakeys',
+        logger: () => {},
+        kmsProviders: {
+          aws: { accessKeyId: 'example', secretAccessKey: 'example' },
+          local: { key: Buffer.alloc(96) }
+        },
+        extraOptions: {
+          csfleSearchPaths: [path.dirname(sharedLibraryStub)]
+        }
+      });
+
+      expect(this.mc._mongocryptdManager).to.equal(undefined);
```

Comment:
Just a small nit: could we use the chai `property` assertion in all the tests here? it makes error logs prettier if this ever fails and differentiates between props whose value is undefined vs those that aren't set at all, which makes the intent clearer:

```js
// if we expect that the prop exists on the object
expect(obj.prop).to.equal('something'); 
// => expected undefined to equal 'something'
expect(obj).to.have.property('prop', 'something'); 
// => if the prop is there:
// 	=> expected obj to have property 'prop' of 'something' but got undefined
// => if the prop was never set:
//	=> expected obj to have property 'prop'

// if we expect that the prop does not exist on the object
expect(obj.prop).to.equal(undefined);
// => expected 'something' to equal undefined
expect(obj).to.not.have.property('prop');
// => expected obj to not have property 'prop'
```

Reply:
Sure, done! :+1: